nacos/src/main/java/com/linecorp/armeria/internal/nacos/LoginClient.java

```diff
@@ -0,0 +1,159 @@
+                                                    long currentTime, long currentDuration) {
+                            return currentDuration;
+                        }
+
+                    })
+                    .buildAsync((key, executor) -> loginInternal());
+
+    LoginClient(HttpClient delegate, WebClient webClient, String username, String password) {
+        super(delegate);
+
+        this.delegate = requireNonNull(delegate, "delegate");
+        this.webClient = requireNonNull(webClient, "webClient");
+        queryParamsForLogin = QueryParams.builder()
+                                         .add("username", requireNonNull(username, "username"))
+                                         .add("password", requireNonNull(password, "password"))
+                                         .toQueryString();
+    }
+
+    private CompletableFuture<AuthToken> login() {
+        return tokenCache.get(NACOS_ACCESS_TOKEN_CACHE_KEY)
+                         .thenApply(loginResult -> loginResult.accessToken);
+    }
+
+    private CompletableFuture<LoginResult> loginInternal() {
+        return webClient.prepare().post("/v1/auth/login")
+                        .content(MediaType.FORM_DATA, queryParamsForLogin)
+                        .asJson(LoginResult.class)
+                        .as(HttpEntity::content)
+                        .execute();
+    }
+
+    @Override
+    public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) {
+        final CompletableFuture<HttpResponse> future = login().thenApply(accessToken -> {
+            try {
+                final HttpRequest newReq = req.mapHeaders(headers -> {
+                    return headers.toBuilder()
+                                  .set(HttpHeaderNames.AUTHORIZATION, accessToken.asHeaderValue())
+                                  .build();
+                });
+                ctx.updateRequest(newReq);
+                return delegate.execute(ctx, newReq);
+            } catch (Exception e) {
+                return Exceptions.throwUnsafely(e);
+            }
+        });
+
+        return HttpResponse.of(future);
+    }
+
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    private static final class LoginResult {
+        private final AuthToken accessToken;
+
+        private final int tokenTtl;
+
+        @Nullable
+        private final Boolean globalAdmin;
+
+        @JsonCreator
+        LoginResult(@JsonProperty("accessToken") String accessToken, @JsonProperty("tokenTtl") Integer tokenTtl,
+                    @JsonProperty("globalAdmin") @Nullable Boolean globalAdmin) {
+            this.accessToken = AuthToken.ofOAuth2(accessToken);
+            this.tokenTtl = tokenTtl;
+            this.globalAdmin = globalAdmin;
+        }
+
+        @Override
+        public String toString() {
+            return MoreObjects.toStringHelper(this)
+                              .omitNullValues()
+                              .add("accessToken", accessToken)
```

Comment:
Changed to use `AuthToken` because sensitive information should not be excluded from serialization.